docs/modules/SDDs/pages/0021-cluster-catalog-compilation.adoc

```diff
@@ -0,0 +1,170 @@
+
+* Cluster ID for which to compile the catalog.
+This can be used in case where a specific cluster should be compiled
+
+* Tenant ID for which to figure out affected clusters.
+This can be used to limit the set of clusters to be considered
+
+* Set of classes that changed
+
+
+==== Authentication & Authorization
+
+To prevent exsessive resource usage or even DoS attacks, the webhooks need to be authenticated.
+A permission system needs to be in place to decide which entities are allowed to trigger which actions.
+It shouldn't be possible for example for a tenant to trigger compilation for clusters of other tenants.
+
+
+=== Compilation Jobs
+
+To compile the catalog for a cluster, a compilation job is created.
+The compilation job is a https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion[Kubernetes job] resource which executes the xref:0006-commodore-configuration-catalog-compilation.adoc[catalog compilation engine].
+
+To distribute the load this compilation generates and to follow a decentralized approach, these jobs are run on the same cluster as they compile the catalog for.
+The Steward component on a cluster starts a compilation job if the `cluster.syn.tools/needs-compilation` annotation is set to `true` on its own cluster object.
+It will then launch the respective job to compile the cluster catalog.
+
+For clusters where this isn't possible due to resource, security or other constraints, a central system runs the jobs.
+These clusters need to be configured in the central system so it will react on the mentioned annotation an create the compile jobs for them.
+
+
+==== Metrics
+
+To properly monitor the compilation jobs, metrics must be generated and stored for each job.
+This can be done by using the Kubernetes metrics already available for the status of each job.
+Additionally, each compilation job pushes detailed metrics about the status to a Prometheus Pushgateway.
+This allows us to create detailed monitoring and alerting rules, even for jobs running on different clusters.
+
+
+=== User Stories
+
+==== Config Change in Global Config Repo
+
+If configuration in the global inventory repo changes, Quartermaster needs to find all clusters which are affected by the changed configuration.
+
+
+==== Config Change in Tenant Config Repo
+
+If configuration in a tenant inventory repo changes, Quartermaster needs to find all clusters which are affected by the changed configuration.
+This will be the same mechanism as for config changes in the global config repo.
+The set of clusters to consider are the clusters of the respective tenant.
+
+
+==== Change of Cluster Facts
+
+If the facts of a cluster change the set of clusters to consider already only consists of this cluster.
+Triggering the webhook needs to be implemented in Lieutenant (operator or API).
+
+
+==== New Release of a Commodore Component
+
+The new release of a Commodore component won't trigger anything directly.
```

Comment:
Do we have already explicitly stated that commodore component will get proper releases and we no longer will just pull latest master? If not, it might be worth be a bit more explicit here. If not part of the discussions, one might not get what is meant by this paragraph.

Reply:
Let's do this in the components docs / how-to write components